packages/beacon-node/src/sync/reconstruction/columnReconstructor.ts

```diff
@@ -0,0 +1,137 @@
+import {toHexString} from "@chainsafe/ssz";
+import {isForkPostFulu} from "@lodestar/params";
+import {RootHex, SignedBeaconBlock, fulu} from "@lodestar/types";
+import {pruneSetToMax} from "@lodestar/utils";
+import {CustodyConfig} from "../../util/dataColumns.js";
+import {GossipedInputType} from "../../chain/blocks/types.js";
+import {IExecutionEngine} from "../../execution/index.js";
+import {
+  getCellsAndProofs,
+  getDataColumnSidecarsFromBlock,
+  getDataColumnSidecarsFromColumnSidecar,
+  kzgCommitmentToVersionedHash,
+} from "../../util/blobs.js";
+import {INetwork} from "../../network/index.js";
+import {SeenGossipBlockInput} from "../../chain/seenCache/seenGossipBlockInput.js";
+import {IBeaconChain} from "../../chain/interface.js";
+
+type ColumnReconstructorBlockInput =
```

Comment:
not sure this name is the best one since reconstuction has entirely different meaning in the context of resconstucting blobs. this seems more like a `local/el fetch` on which lines it could be named